### DISCUSSION TITLE
Pyodide online console relies on pyodide-lock.json from version 0.26.1, can it be updated so new packages are added?

### DISCUSSION THREAD

**Discussion by carlopi** [pyodide/pyodide#4953] [General]
@cpcloud recently added the duckdb package (https://github.com/pyodide/pyodide/pull/4684), how to get that as part of the console distribution?

I was looking around, and I am under the impression that the packages are frozen as to the release 0.26.1 (given the file is fetched from cdn tied to the npm release).

It's possible to have the package list updated more frequently ? Both new versions and new packages can benefit from that.

I am sorry if this is common knowledge, discussed elsewhere.

Thanks a lot, super cool project,

Carlo, part of the DuckDB team

**Comment by ryanking13** [comment]
Each Pyodide release includes a set, pre-calculated list of packages (pyodide-lock.json). However, you are not limited to this set of packages (although they are slightly more inconvenient to use).

1. You can install the package from the URL using `micropip.install('https://...')` or `pyodide.loadPackage('https://...')`.
2. You can use a different lockfile that includes duckdb. For instance, you can download Pyodide distribution from our [latest CI result](https://app.circleci.com/pipelines/github/pyodide/pyodide/7563/workflows/637caf8b-366c-4278-861e-a9cb93adb344/jobs/108127/artifacts), host it somewhere, and change the [`lockFileURL` parameter when loading Pyodide](https://pyodide.org/en/stable/usage/api/js-api.html#globalThis.loadPyodide).

But I agree that this isn't convenient enough, and we're looking for a better way. However, this has something to do with Python's overall packaging system and compatibility issues with Emscripten than with Pyodide itself, so it's unlikely that this will be resolved soon.

---

Another simple and possible way would be to use our latest console (https://pyodide.org/en/latest/console.html) until we release the next version of Pyodide. The latest console is deployed on every commit.
